src/cudadev/CUDADataFormats/SiPixelDigisCUDA.cc

```diff
@@ -5,44 +5,54 @@
 #include "CUDACore/host_unique_ptr.h"
 
 SiPixelDigisCUDA::SiPixelDigisCUDA(size_t maxFedWords, cudaStream_t stream)
-    : xx_d(cms::cuda::make_device_unique<uint16_t[]>(maxFedWords, stream)),
-      yy_d(cms::cuda::make_device_unique<uint16_t[]>(maxFedWords, stream)),
-      adc_d(cms::cuda::make_device_unique<uint16_t[]>(maxFedWords, stream)),
-      moduleInd_d(cms::cuda::make_device_unique<uint16_t[]>(maxFedWords, stream)),
-      clus_d(cms::cuda::make_device_unique<int32_t[]>(maxFedWords, stream)),
-      view_d(cms::cuda::make_device_unique<DeviceConstView>(stream)),
-      pdigi_d(cms::cuda::make_device_unique<uint32_t[]>(maxFedWords, stream)),
-      rawIdArr_d(cms::cuda::make_device_unique<uint32_t[]>(maxFedWords, stream)) {
-  auto view = cms::cuda::make_host_unique<DeviceConstView>(stream);
-  view->xx_ = xx_d.get();
-  view->yy_ = yy_d.get();
-  view->adc_ = adc_d.get();
-  view->moduleInd_ = moduleInd_d.get();
-  view->clus_ = clus_d.get();
-
-  cms::cuda::copyAsync(view_d, view, stream);
-}
+    : data_d(cms::cuda::make_device_unique<std::byte[]>(
+        DeviceOnlyStore::computeDataSize(maxFedWords) + 
+        HostDeviceStore::computeDataSize(maxFedWords),
```

Comment:
Is the return value of `computeDataSize()` esseitnally rounded up to the next integer multiple of alignment bytes?

Reply:
A bit more. It computes the sizes of each column, which have to be individually aligned and sums the sizes. On example can be found here: https://gist.github.com/ericcano/1d10e0902829408a0d8815e99567102c#file-cc-L19.